pkg/cloud/services/eks/nodegroup.go

```diff
@@ -496,14 +546,18 @@ func (s *NodegroupService) reconcileNodegroup() error {
 	switch *ng.Status {
 	case eks.NodegroupStatusCreating, eks.NodegroupStatusUpdating:
 		ng, err = s.waitForNodegroupActive()
+		if err != nil {
+			return errors.Wrap(err, "failed to wait for nodegroup to be active")
+		}
+		conditions.MarkTrue(s.scope.ManagedMachinePool, expinfrav1.EKSNodegroupReadyCondition)
+		if conditions.IsFalse(s.scope.ManagedMachinePool, expinfrav1.EKSNodegroupUpdateSucceededCondition) &&
```

Comment:
Maybe we could just set the `EKSNodegroupUpdateSucceededCondition` to true if we get here and if `*ng.Status==eks.NodegroupStatusUpdating`? wdyt?

Reply:
I think the status will have changed by the time we get to this line, because we are calling `ng, err := s.waitForNodegroupActive()` which looks like it will update `ng.Status` to something else: https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/f4c22203805d14b5bc57ca7605a9ee611d175e76/pkg/cloud/services/eks/nodegroup.go#L640-L646

Reply:
Thats a good point @jon-fearer, i missed that...thanks :)

The intent of the condition check logic on [552-555 ](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/cec813cf16c54cd1e3570926cc50342f44f9d7a0/pkg/cloud/services/eks/nodegroup.go#L552:L555) isn't immediately obvious to me on reading the code. 

I'm wonder if one of the following would make it more obvious:
*  Extract the `conditions.IsFalse(s.scope.ManagedMachinePool, expinfrav1.EKSNodegroupUpdateSucceededCondition) && conditions.GetReason(s.scope.ManagedMachinePool, expinfrav1.EKSNodegroupUpdateSucceededCondition) == expinfrav1.EKSNodegroupUpdatingReason` into a function with a name that indicates its intent
* Split the ` eks.NodegroupStatusCreating, eks.NodegroupStatusUpdating` into 2 separate cases in the switch statement, with the `eks.NodegroupStatusUpdating` case doing the additional `conditions.MarkTrue(s.scope.ManagedMachinePool, expinfrav1.EKSNodegroupUpdateSucceededCondition)`. We'd probably want to extra the common code into a function that is called in each case.

wdyt?

Reply:
I'm wondering if those if statements are even necessary. I think the main reason for putting them in there was wanting to avoid setting the `EKSNodegroupUpdateSucceededCondition` to true when the node group is created for the first time (since it is a "create" and not an "update"). But if that's ok, then always marking them both true when the node group is active should be fine, so I just moved both `markTrue(...)` statements into a separate function. Does that behavior make sense to you (always setting both conditions to true when cluster becomes active)?

Reply:
If we only want to set the`EKSNodegroupUpdateSucceededCondition` to true on updates (and not creates), then I could do something like this instead:
```
// markReadyAndUpdateConditionsTrue sets the EKSNodegroupReadyCondition and EKSNodegroupUpdateSucceededCondition
// to true for the ManagedMachinePool.
func (s *NodegroupService) markReadyAndUpdateConditionsTrue() {
	conditions.MarkTrue(s.scope.ManagedMachinePool, expinfrav1.EKSNodegroupReadyCondition)
	if (s.isNodegroupConditionUpdating()) {
		conditions.MarkTrue(s.scope.ManagedMachinePool, expinfrav1.EKSNodegroupUpdateSucceededCondition)
	}
}

func (s *NodegroupService) isNodegroupConditionUpdating() bool {
	return conditions.IsFalse(s.scope.ManagedMachinePool, expinfrav1.EKSNodegroupUpdateSucceededCondition) &&
		conditions.GetReason(s.scope.ManagedMachinePool, expinfrav1.EKSNodegroupUpdateSucceededCondition) == expinfrav1.EKSNodegroupUpdatingReason
}
```

Reply:
> then always marking them both true when the node group is active should be fine, so I just moved both `markTrue(...)` statements into a separate function. Does that behavior make sense to you (always setting both conditions to true when cluster becomes active)

Thats a good question :+1:  I don't think we should set `expinfrav1.EKSNodegroupUpdateSucceededCondition` to true on
create as this gives the impression that an update has been done when it hasn't.

Reply:
Ok sounds good. Here are those [changes](https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/3027/files#diff-ed6ead8e5e6ab86b4693e189f4568484aeda9b9c660e21f70f08fab9f5da13e3R652-R666). Let me know your thoughts